libs/sdk/src/RudderClient.ts

```diff
@@ -280,6 +281,10 @@ async function putAnonymousId(anonymousId: string) {
 }
 
 async function reset(clearAnonymousId = false) {
+  if (typeof clearAnonymousId !== 'boolean') {
+    logWarn("reset: 'clearAnonymousId' must be a boolean");
+    return;
+  }
   bridge.reset(clearAnonymousId);
```

Comment:
Instead of returning from here, we can get the boolean value of the input and proceed as we're rightly warning the user.

```suggestion
  const clearAnonymousIdVal = clearAnonymousId === true;
  if (typeof clearAnonymousId !== 'boolean') {
    logWarn("reset: 'clearAnonymousId' must be a boolean");
  }
  bridge.reset(clearAnonymousIdVal);
```